client/src/dojo/modelManager/BattleManager.ts

```diff
@@ -124,14 +147,147 @@ export class BattleManager {
               ),
             ),
           );
+
+    cloneArmy.health.current = this.getTroopFullHealth(cloneArmy.troops);
+
     return cloneArmy;
   }
 
-  private getTroopFullHealth(troops: Troops): bigint {
+  public isClaimable(
+    currentTimestamp: number,
+    selectedArmy: ArmyInfo | undefined,
+    structure: Structure | undefined,
+    defender: ArmyInfo | undefined,
+  ): boolean {
+    if (!selectedArmy) return false;
+    if (this.battleIsClaimable) return this.battleIsClaimable;
+
+    if (this.isBattleOngoing(currentTimestamp)) {
+      return false;
+    }
+
+    if (!structure) {
+      this.battleIsClaimable = false;
+      return false;
+    }
+    if (this.getBattleType(structure) !== BattleType.Structure) {
+      this.battleIsClaimable = false;
+      return false;
+    }
```

Comment:
remove if (!structure) condition, 
it is already checked within the if (this.getBattleType(structure) !== BattleType.Structure) check